domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/ShapePoint.java

```diff
@@ -0,0 +1,240 @@
+        private Float shapeDistTraveled;
+
+        /**
+         * Sets field shapeId value and returns this
+         *
+         * @param shapeId identifies a shape
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapeId(@NotNull final String shapeId) {
+            this.shapeId = shapeId;
+            return this;
+        }
+
+        /**
+         * Sets field shapePtLat value and returns this
+         *
+         * @param shapePtLat latitude of a shape point
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapePtLat(@NotNull final Float shapePtLat) {
+            this.shapePtLat = shapePtLat;
+            return this;
+        }
+
+        /**
+         * Sets field shapePtLon value and returns this
+         *
+         * @param shapePtLon longitude of a shape point
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapePtLon(@NotNull final Float shapePtLon) {
+            this.shapePtLon = shapePtLon;
+            return this;
+        }
+
+        /**
+         * Sets field shapePtSequence value and returns this
+         *
+         * @param shapePtSequence sequence in which the shape points connect to form the shape
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapePtSequence(@NotNull final Integer shapePtSequence) {
+            this.shapePtSequence = shapePtSequence;
+            return this;
+        }
+
+        /**
+         * Sets field shapeDistTraveled value and returns this
+         *
+         * @param shapeDistTraveled actual distance traveled along the shape from the first shape point to the point
+         *                          specified in this record
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapeDistTraveled(@Nullable final Float shapeDistTraveled) {
+            this.shapeDistTraveled = shapeDistTraveled;
+            return this;
+        }
+
+        /**
+         * Entity representing a row from shapes.txt if the requirements from the official GTFS specification
+         * are met. Otherwise, method returns an entity representing a list of notices.
+         *
+         * @return Entity representing a row from shapes.txt if the requirements from the official GTFS specification
+         * are met. Otherwise, method returns an entity representing a list of notices.
+         */
+        public EntityBuildResult<?> build() {
+            noticeCollection.clear();
+
+            // suppressed warning regarding nullability of fields shapePtLat, shapePtLon, and shapePtSequence since
+            // these fields could be provided as null values
+            //noinspection ConstantConditions to avoid lint
+            if (shapeId == null ||
+                    shapePtLat == null || (shapePtLat != null && (shapePtLat < -90.0f || shapePtLat > 90.0f)) ||
```

Comment:
The `shapePtLat < -90.0f || shapePtLat > 90.0f` and `shapePtLon < -180.0f || shapePtLon > 180.0f` should be encapsulated in their own methods for `isValidLatitude()` and `isValidLongitude()` accordingly. You'll need these for stops.txt too.